src/zimwriterfs/zimwriterfs.cpp

```diff
@@ -346,6 +349,26 @@ void create_zim()
 {
   ZimCreatorFS zimCreator(directoryPath, welcome, isVerbose(), uniqueNamespace, zstdFlag);
 
+  if (zimPath.size() >= (MAXPATHLEN-1))
+    throw std::invalid_argument("Target .zim file path is too long");
+
+  // dirname() can modify the 'buf', no c_str() here.
+  char buf[MAXPATHLEN];
+  strncpy(buf, zimPath.c_str(), sizeof(buf));
+  std::string zimdir = dirname(buf);
+
+  if (zimdir.size() >= MAXPATHLEN)
+    throw std::invalid_argument("Target .zim directory path is too long");
+
+  if (realpath(zimdir.c_str(), buf) != buf)
```

Comment:
I dislike `if` test without `{}`. Especially if there are several lines after it.

I prefer to always have the coding style :

```
if (foo) {
    bar;
}
```

Even if there is only one instruction in the `if`.